substrate/frame/examples/basic/src/lib.rs

```diff
@@ -82,6 +96,8 @@ pub use weights::*;
 
 /// A type alias for the balance type from this pallet's point of view.
 type BalanceOf<T> = <T as pallet_balances::Config>::Balance;
+
+/// A way to calculate transaction fees for our custom weight calculator.
 const MILLICENTS: u32 = 1_000_000_000;
 
 // A custom weight calculator tailored for the dispatch call `set_dummy()`. This actually examines
```

Comment:
A pretty useful step is to also remove `WeightForSetDummy` and everything around it. We never implement this `WeighData` manually.

Reply:
Instead, provide a reasonable and concise intro into what weights are, else keep them a constant for now. 

Or make the pallet `dev_mode`.

In any case, this `WeightForSetDummy` is noise though in my opinion.

Reply:
Yeah I agree - this was some stuff left from a long time ago. It could make for some good learning material about how weights work eventually - in particular implementing ClassifyDispatch and PaysFee. Though here it is noise. I'll remove it. This will also drastically simplify the pallet!

Reply:
In the last commits I removed all custom weight stuff. Now I'm unsure whether to:
1. Provide proper benchmarked weights for `set_dummy` 
2. OR make the pallet `dev_mode`

We already have a pallet that show cases dev_mode and I like how this example pallet provides the correct way for setting weights, plus using the macro for `Inherited weight annotation for pallet calls` which we show can be overriden, so I currently just set the weight for `set_dummy` to `0` to show this - though I'm weary about promoting this even as an example as it's not something any runtime dev should ever do.

Wdyt @kianenigma  ?